docker-examples/poetry-multistage/Dockerfile

```diff
@@ -0,0 +1,38 @@
+FROM python:3.11-slim as builder
+
+# --- Install Poetry ---
+ARG POETRY_VERSION=1.8
+
+ENV POETRY_HOME=/opt/poetry
+ENV POETRY_NO_INTERACTION=1
+ENV POETRY_VIRTUALENVS_IN_PROJECT=1
+ENV POETRY_VIRTUALENVS_CREATE=1
+ENV PYTHONDONTWRITEBYTECODE=1
+ENV PYTHONUNBUFFERED=1
+# Tell Poetry where to place its cache and virtual environment
+ENV POETRY_CACHE_DIR=/opt/.cache
+
+RUN pip install "poetry==${POETRY_VERSION}"
+
+WORKDIR /app
+
+# --- Reproduce the environment ---
+# You can comment the following two lines if you prefer to manually install
+#   the dependencies from inside the container.
+COPY pyproject.toml .
```

Comment:
Shouldn't we copy also copy `poetry.lock` for reproducibility?

Reply:
Done, like for https://github.com/python-poetry/poetry/pull/9542#discussion_r1797755173. Thanx